package/com.unity.formats.usd/Runtime/Scripts/IO/Materials/ShaderExporterBase.cs

```diff
@@ -33,33 +33,120 @@ public class ShaderExporterBase {
-#else
-      // Not supported at run-time, too many things can go wrong
-      // (can't encode compressed textures, etc).
-      throw new System.Exception("Not supported at run-time");
-#endif
-    }
+                                 string textureOutput)
+    {
+      // We have to handle multiple cases here:
+      // - file exists on disk
+      //   - file is a supported format => can be directly copied
+      //   - file is not in a supported format => need to blit / export
+      // - file is only in memory
+      //   - a Texture2D
+      //   - a Texture
+      //   - a RenderTexture
+      //   - needs special care if marked as Normal Map
+      //     (can probably only be detected in an Editor context, and heuristically at runtime)
+      //   => need to blit / export
+      // - file is not supported at all (or not yet)
+      //   - a 3D texture
+      //   => needs to be ignored, log Warning
+
+      bool textureIsExported = false;
+
+      string filePath = null;
+      string fileName = null;
+
+      var texture = material.GetTexture(textureName);
+
+  #if UNITY_EDITOR
+      var srcPath = UnityEditor.AssetDatabase.GetAssetPath(texture);
+      if(!string.IsNullOrEmpty(srcPath))
+      { 
+        srcPath = srcPath.Substring("Assets/".Length);
+        srcPath = Application.dataPath + "/" + srcPath;
+        fileName = System.IO.Path.GetFileName(srcPath);
+        filePath = System.IO.Path.Combine(destTexturePath, fileName);
+
+        if(System.IO.File.Exists(srcPath))
+        { 
+          // USD officially only supports png / jpg / jpeg
+          // https://graphics.pixar.com/usd/docs/Usdz-File-Format-Specification.html
 
+          var ext = System.IO.Path.GetExtension(srcPath).ToLowerInvariant();
+          if(ext == ".png" || ext == ".jpg" || ext == ".jpeg") { 
+            System.IO.File.Copy(srcPath, filePath, overwrite: true);
+            if (System.IO.File.Exists(filePath))
+              textureIsExported = true;
+          }
+        }
+      }
+  #endif
+      if (!textureIsExported)
+      {
+        fileName = texture.name + "_" + Random.Range(10000000, 99999999).ToString();
+  #if UNITY_EDITOR
+        if (UnityEditor.AssetDatabase.Contains(texture))
+          fileName = texture.name + "_png";
+  #endif
+        filePath = System.IO.Path.Combine(destTexturePath, fileName + ".png");
+        
+        // TODO extra care has to be taken of Normal Maps etc., since these are in a converted format in memory (namely, 16 bit G and A instead of 8 bit RGBA)
+        // An example of this conversion in a shader is in Khronos' UnityGLTF implementation.
+        // basically, the blit has do be done with the right unlit conversion shader to get a proper "file-based" tangent space normal map back
+
+        // Blit the texture and get it back to CPU
+        // Note: Can't use RenderTexture.GetTemporary because that doesn't properly clear alpha channel
```

Comment:
It's probably a bad idea to rely on any RenderTexture to be initialized to zeros, especially if the pipeline is configured to be a modern explicit graphics API.

Reply:
added a GL.Clear